frontend/src/staging/FileBrowser.vue

```diff
@@ -1,40 +1,245 @@
+                    <b-table-column label="">
+                        <b-button v-if="props.row.name !== parentFolderName"
+                                  icon-right="delete" @click="deleteFile(props.row)" />
+                    </b-table-column>
+                </template>
+                 <template slot="detail" slot-scope="props">
+                     <p>{{ props.row.name }}</p>
+                 </template>
+            </b-table>
+        </div>
+        <div v-else>
+            No files found!
+        </div>
+    </section>
 </template>
 
 <script lang="ts">
 import {
-    Component, Prop, Vue, Watch
+    Component, Prop, Vue, Watch, Inject
 } from 'vue-property-decorator';
+import { showSuccess, showWarning, showError } from '@/util/Notifier.ts';
+
+const parentFolderName: string = '..';
 
 @Component
 export default class FileBrowser extends Vue {
-    @Prop() filesToShow!: object[]
-    @Prop() initialSelected!: object
+    @Inject() fetchFunction!: Function
+    @Inject() uploadFunction!: Function
+    @Inject() deleteFunction!: Function
+    @Inject() createFolderFunction!: Function
 
-    columns: object[] = [
-        {
-            field: 'name',
-            label: 'Filename'
-        }
-    ]
+    @Prop() initialSelected!: WorkbenchFile
+
+    // needs to be on component to be usable in template
+    parentFolderName: string = parentFolderName;
+    getFolderIcon: Function = getFolderIcon;
 
-    selected = this.initialSelected || {};
+    selectedFile: WorkbenchFile = this.initialSelected || null;
+    workingDirectory: string = ''
 
-    @Watch('selected')
-    onSelectedChanged(value: string, oldValue: string) {
-        this.$emit('update:initialSelected', this.selected);
+    filesToShow: WorkbenchFile[] = []; // TODO type
+
+    uploadRunning: boolean = false;
+    filesToUpload: File[] = [];
+
+    createFolder(folderPath: string) {
+        this.$buefy.dialog.prompt({
+            message: `Enter the folder name or path`,
+            inputAttrs: {
+                maxlength: 20
+            },
+            onConfirm: (folderName) => {
+                this.createFolderFunction(this.getFilePath(folderName));
+                this.fetchFiles();
+            }
+        });
+    }
+
+    mounted() {
+        this.fetchFiles();
+    }
+
+    async fetchFiles() {
+        try {
+            this.filesToShow = await this.fetchFunction();
+            this.filesToShow.sort(compareFileEntries);
+
+            // go to original folder
+            const oldWorkingDir = this.workingDirectory;
+            const folderArray = this.workingDirectory.split('/');
+            folderArray.shift(); // ignore root folder
+            folderArray.forEach((item) => {
+                this.openFolder(item);
+            });
+            this.workingDirectory = oldWorkingDir;
+        } catch (e) {
+            showError('Failed to retrieve file list from server!', e);
+        }
+    }
+
+    @Watch('selectedFile')
```

Comment:
Die Konstruktion war vorher schon da, ich kommentiere es trotzdem mal ...

Ich verstehe gerade nicht, warum hier `Watch` notwendig ist. Wenn ich das richtig sehe, wird hier auf Änderungen von `selectedFile` reagiert, dann ein Event emittet, das sagt, dass `initialSelected` sich auf den Wert von `selectedFile` geändert hat, nur um dann im nächsten `Watch` `selectedFile` auf den Wert zu setzen, den es eh schon hat.

Aus meiner Sicht, wäre es viel einfacher, `selectedFile` zu einer `Prop` zu machen und `initialSelected` wegzuwerfen. `Watch` wäre dann auch nicht nötig, weil sich Vue um das binding kümmert.

Oder habe ich da etwas übersehen?

Reply:
Ja ich finde auch ein bischen umständlich. Das Problem mit deiner vorgeschlagenen Variante ist, dass dann in der Console folgender Fehler erscheint:

[Vue warn]: Avoid mutating a prop directly since the value will be overwritten whenever the parent component re-renders. Instead, use a data or computed property based on the prop's value. Prop being mutated: "initialSelected"

Man muss also immer eine lokale Kopie von Props anlegen, wenn man die verändern will.
Weiss nicht, ob ich das vollumfänglich verstanden habe. Also wenn du da konkrete Vorschläge hast, immer her damit. Bzw mach es selbst. ;)